### ISSUE TITLE
linter: incorrect defaults for typescript/no-expliciy-any

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#12561]
> I think this is the same for typescript/no-explicit-any, it should be enabled with the default config "error" (https://typescript-eslint.io/rules/no-explicit-any/) but when activating the plugin and adding the restriction category it does not reports errors, it needs to be configured manually. 

 _Originally posted by @J3m5 in [#12532](https://github.com/oxc-project/oxc/issues/12532#issuecomment-3122117340)_

**Comment by camc314** [comment]
@J3m5 could you share a bit more detail?

I tried to reproduce this here:

https://github.com/camc314/oxlint-repros/tree/main/issue-12561

but when i run oxlint, i see

```
pnpm run lint

> @ lint /Users/cameron/github/camc314/oxlint-repros/issue-12561
> oxlint


  × typescript-eslint(no-explicit-any): Unexpected any. Specify a different type.
   ╭─[src/index.ts:1:24]
 1 │ export function foo(): any {
   ·                        ───
 2 │     throw new Error('This function is not implemented yet.');
   ╰────
  help: Use `unknown` instead, this will force you to explicitly, and safely, assert the type is correct.

Found 0 warnings and 1 error.
Finished in 6ms on 1 file using 12 threads.
 ELIFECYCLE  Command failed with exit code 1.
```


Which should be the expected behaviour

**Comment by J3m5** [comment]
I tested it again and it's actually working as expected, maybe it didn't in a previous release or I confused it with another rule

**Comment by camc314** [comment]
Thanks for double checking
